diesel/src/connection/mod.rs

```diff
@@ -24,22 +24,23 @@ pub trait SimpleConnection {
     ///
     /// This function is used to execute migrations,
     /// which may contain more than one SQL statement.
-    fn batch_execute(&self, query: &str) -> QueryResult<()>;
+    fn batch_execute(&mut self, query: &str) -> QueryResult<()>;
 }
 
 /// A connection to a database
-pub trait Connection: SimpleConnection + Send {
+pub trait Connection: SimpleConnection + Sized + Send {
     /// The backend this type connects to
     type Backend: Backend;
 
+    #[doc(hidden)]
```

Comment:
It was hidden before we removed it. So I've opted to hide it again as we now just added that type back. This will probably be available as part of the `I-Implement-a-third-party-backend` feature as outlined in #2696